app/Services/YouTubeClient.php

```diff
@@ -20,6 +20,8 @@ public function channel(string $id): ChannelData
 
     public function channels(iterable|string $channelIds): Collection
     {
+        $channelIds = is_array($channelIds) ? $channelIds : [$channelIds];
```

Comment:
I've noticed this issue as well, it seems Larastan is not happy with a string being passed to `collect()`. I will talk to Can about that.

Reply:
PS: for context, @SamuelMwangiW added this line to make sure `$channelIds` is an array. Besides Larastan, it also works if it is a string.

Reply:
@christophrumpel Yes indeed `collect()` works perfectly when a string is passed to it however the method [signature in Laravel](https://github.com/laravel/framework/blob/13372872bed31ae75df8709b9de5cde01d50646e/src/Illuminate/Collections/helpers.php#L13) does not include string as a param type which I guess is why Larastan complains. 
As per the underlying [EnumeratesValues trait](https://github.com/laravel/framework/blob/13372872bed31ae75df8709b9de5cde01d50646e/src/Illuminate/Collections/Traits/EnumeratesValues.php#L992), Laravel casts values to array by default hence this is unnecessary

Reply:
Larastan actually supports passing scalar values to `collect` https://github.com/nunomaduro/larastan/blob/master/src/ReturnTypes/Helpers/CollectExtension.php

But yeah I also think the typehint in `collect` function in Laravel causes this issues. And this can be overridden in Larastan. I'll try to do that for next version 👍🏽

Reply:
Ah, I just saw the build failure here: https://github.com/christophrumpel/larastreamers/runs/5234051729?check_suite_focus=true#step:7:28

This looks like it's not related to `string`, but `iterable` 

Can you revert this change and add a PHPDoc for the `$channelIds` that looks like this `@param iterable<int, string>|string $channelIds`? And see if that also fixes it.

Reply:
Thanks, @canvural for taking a look and working on that 🙏🏻

Regarding "Larastan actually supports passing scalar values to collect", does this mean we can actually make Larastan already happy before the changes you will look into?

Reply:
> Ah, I just saw the build failure here: https://github.com/christophrumpel/larastreamers/runs/5234051729?check_suite_focus=true#step:7:28
> 
> This looks like it's not related to `string`, but `iterable`
> 
> Can you revert this change and add a PHPDoc for the `$channelIds` that looks like this `@param iterable<int, string>|string $channelIds`? And see if that also fixes it.

I have reverted the commit and applied your suggestion but still getting a similar error for my local build and seems the same is true in the github actions run. 
<img width="979" alt="image" src="https://user-images.githubusercontent.com/1807304/155291080-a19ce148-6f85-4f31-a0dc-44b44ce2302e.png">

Reply:
Ah, that's unfortunate 🙁 So I guess this needs to be fixed in Larastan. So feel free to ignore this error, or like previous you can add a check with `is_array`

Reply:
Thanks @canvural for your assistance and the amazing work you do. I have reverted back to https://github.com/christophrumpel/larastreamers/commit/1a8697b94293b50470d784ee20ec6163f9851f54 just to leave the PR in a passing state.